### PR TITLE
Moved ipython to requirements.txt

### PR DESCRIPTION
**Pull request by giocalitri**
#### What are the relevant tickets?
N/A

#### What's this PR do?
Moves ipython to requirements.txt
It also removes the test_requirements.in (no more pinned test requirements) and sets the pip cache environment variable to `/src/.cache`

#### How should this be manually tested?
1- the tests should pass on travis.
2- connect to the heroku build for this PR with the `./manage.py shell` and verify that ipython has been used (instead of the default python)

### CODE REVIEW

**Comment by noisecapella** [comment]
Looks like the test is complaining about a double requirement

File: requirements.txt [hunk 1]
```diff
@@ -32,21 +34,30 @@ elasticsearch-dsl==2.1.0
 elasticsearch==2.3.0
 factory-boy==2.8.1
 faker==0.7.3
-html5lib==0.999999999
+html5lib==0.999999
```
Comment (noisecapella) [inline requirements.txt]: This looks accidental, can you revert?

File: test_requirements.txt [hunk 2]
```diff
@@ -25,7 +25,7 @@ first==2.0.1              # via pip-tools
 greenlet==0.4.11          # via bpython
 ipdb==0.10.1
 ipython-genutils==0.1.0   # via traitlets
-ipython==5.1.0
+ipython==5.1.0            # via ipdb
```
Comment (noisecapella) [inline test_requirements.txt]: Shoudl we move over ipdb too?

**Comment by giocalitri** [comment]
Yep
this PR also shows that `pyup did not modify the right file

File: requirements.txt (hunk 1 above)
Comment (giocalitri) [inline requirements.txt]: this is because yup modified the wrong file: I will fix it

File: test_requirements.txt (hunk 2 above)
Comment (giocalitri) [inline test_requirements.txt]: ipdb is a debugger: I do not see the reason to move it. 
maybe I can keep ipython here with a strict requirement. I'll test something

**Comment by noisecapella** [comment]
It looks like the selenium failures are actually pylint failures in the pylint directory, maybe because it's not using the same `mm_travis_web` image. Can you rebase on master and see if the problem persists?

**Comment by giocalitri** [comment]
@noisecapella I rebased and this seems to have fixed the selenium tests

File: .travis.yml [hunk 3]
```diff
@@ -7,6 +7,7 @@ matrix:
     - install:
       - env | grep TRAVIS > .env
       - env | grep CI >> .env
+      - echo "XDG_CACHE_HOME=/src/.cache" >> .env
```
Comment (noisecapella) [inline .travis.yml]: Why not put this in `travis-docker-compose.yml`? I'm fine either way but just wondering

Comment (giocalitri) [inline .travis.yml]: no reson: I will move it

Comment (giocalitri) [inline .travis.yml]: it is probably one of the million tests I did on friday

**Pull request merged**